### DISCUSSION TITLE
imgproxy

### DISCUSSION THREAD

**Discussion by tinect** [FriendsOfShopware/FroshPlatformThumbnailProcessor#36] [Patterns]
These patterns are for selfhosted https://imgproxy.net/

From imgproxy 3.0.0:
```
http://localhost:8080/_/rt:fit/w:{width}/h:0/g:sm/plain/{mediaUrl}/{mediaPath}
```

till imgproxy 3.0.0:
```
http://localhost:8080/x/fit/{width}/0/sm/0/plain/{mediaUrl}/{mediaPath}
```

Replace `http://localhost:8080` with your hosted address.

**Comment by Max-SBYP** [comment]
The pattern is correct, but shopware renames ":" to %3A.

**Reply by tinect** [reply]
**EDIT: this has been fixed.**

Sadly, yes. But we can manage this through plugin.
Here is a proof-of-concept: https://github.com/FriendsOfShopware/FroshPlatformThumbnailProcessor/pull/41/

But I would like to improve it... maybe through PHP.~

**Reply by Timothy0** [reply]
I think the problem is back, since the frosh_encode_url twig filter was removed after v1.1.2. Or am i missing someting maybe?

**Reply by tinect** [reply]
Yes. It has been removed from this plugin. Please use FroshLazySizes or implement it in your project on you own.

**Comment by Isengo1989** [comment]
@tinect do you have any experience on how to scale imgproxy lets say for 1TB of traffic per month? Before trying I would like to estimate the cost for a VM compared to lets say bunnys image service (9USD/Month -> would mean a small VM on DigitalOcean)

**Reply by tinect** [reply]
Sadly not :-(
You really mean 1 TB traffic ONTO imgproxy?

But, the requests should not go onto imgproxy directly, but cached in a proxy/cdn.
Once the images are converted they should be taken from the proxy and not re-requested from imgproxy.
